examples/C++/DDS/HelloWorldExample/HelloWorldSubscriber.cpp

```diff
@@ -125,29 +192,41 @@ void HelloWorldSubscriber::SubListener::on_data_available(
         DataReader* reader)
 {
     SampleInfo info;
-    if (reader->take_next_sample(&hello_, &info) == ReturnCode_t::RETCODE_OK)
+    while (reader->take_next_sample(&hello_, &info) == ReturnCode_t::RETCODE_OK)
```

Comment:
Study if this loop is required.
`on_data_available` is called once for each data arrived, even when they arrive together.
Trying with high frequencies (-i 1) it never repeats this `while` loop

Reply:
:+1:

Reply:
Sorry, y fault. This should be a while just in case. It should not happend that a data received does not raise a callback, but it could happend

Reply:
:+1: